src/utils/io/progress_log.rs

```diff
@@ -183,17 +197,37 @@ impl Progress {
     fn emit_log_if_required(&mut self) {
         let now = Instant::now();
         if (now - self.last_logged) > UPDATE_FREQUENCY {
+            let msg = self.msg(now);
+            if let Some(cb) = &self.callback {
+                cb(msg.clone());
+            }
+
             tracing::info!(
                 target: "forest::progress",
                 "{}",
-                self.msg(now)
+                msg
             );
             self.last_logged = now;
             self.last_logged_items = self.completed_items;
         }
     }
 }
 
+// Derive Debug for Progress, skipping the callback field
+impl std::fmt::Debug for Progress {
```

Comment:
Alternatively, https://crates.io/crates/educe can be used to ignore the `callback` field

Reply:
Done.

Reply:
Thanks